src/app/client/src/app/modules/groups/components/group-details/group-details.component.ts

```diff
@@ -92,10 +92,15 @@ export class GroupDetailsComponent implements OnInit, OnDestroy {
     this.showFilters = true;
   }
 
-  handleNextClick(event) {
-    this.toggleActivityModal(false);
-    this.addActivityModal.deny();
-    this.router.navigate([`${ADD_ACTIVITY_TO_GROUP}/${COURSES}`, 1], { relativeTo: this.activatedRoute });
+  navigateToAddActivity() {
+   this.router.navigate([`${ADD_ACTIVITY_CONTENT_TYPES}`], {
+     relativeTo: this.activatedRoute,
+     queryParams: {
+       groupName: _.get(this.groupData, 'name'),
+       createdBy: _.capitalize(_.get(_.find(this.groupData['members'], {userId: this.groupData['createdBy']}), 'name')),
+       groupId: _.get(this.groupData, 'id'),
```

Comment:
This is the must parameter, Through error when this.groupData.id is not available. Don't navigate also.

Reply:
not required. No need to pass groupId in the query params, I have changed the code. while setting up the pageContext, groupId is must,